### ISSUE TITLE
AT commands through USB on ESP-DevKitC. No respose.

### ISSUE THREAD

**Issue by TomaszKosecki** [espressif/esp32-at#153]
Hi,

Hardware:
ESP32-DevKitC with ESP32-WROOM-32D onboard.
Firmware:
This repo master branch commit: 4cbb1e3

After flashing this firmware and reseting device i get "ready" on terminal but i don't receive any responses for AT commands e.g. I send:
AT+GMR
and there is no answer.

Terminal log :  (I use Tera Term on Windows 10)
```
ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:6272
ho 0 tail 12 room 4
load:0x40078000,len:10168
ho 0 tail 12 room 4
load:0x40080400,len:6468
entry 0x40080758

ready

```

I know that by default AT commands use different UART than the one which is wired to USB-UART converter so I changed setting using "make menuconfig" to these:
(important parts of my "sdkconfig" file)

```
CONFIG_CONSOLE_UART_DEFAULT=
CONFIG_CONSOLE_UART_CUSTOM=
CONFIG_CONSOLE_UART_NONE=y
CONFIG_CONSOLE_UART_NUM=0

```
```

 #
# AT uart settings
#
CONFIG_AT_UART_PORT=0
CONFIG_AT_UART_PORT_RX_PIN=3
CONFIG_AT_UART_PORT_TX_PIN=1
CONFIG_AT_UART_PORT_RTS_PIN=14
CONFIG_AT_UART_PORT_CTS_PIN=15
CONFIG_AT_UART_DEFAULT_BAUDRATE=115200
CONFIG_AT_UART_DEFAULT_DATABITS=8
CONFIG_AT_UART_DEFAULT_STOPBITS=1
CONFIG_AT_UART_DEFAULT_PARITY_BITS=0
CONFIG_AT_UART_DEFAULT_FLOW_CONTROL=0
CONFIG_AT_BASE_ON_HSPI=
CONFIG_AT_BASE_ON_SOCKET=
CONFIG_AT_BASE_ON_SDIO=
CONFIG_AT_COMMAND_TERMINATOR_SUPPORT=
CONFIG_AT_BASE_COMMAND_SUPPORT=y
CONFIG_AT_WIFI_COMMAND_SUPPORT=y
CONFIG_AT_NET_COMMAND_SUPPORT=y
CONFIG_AT_BLE_COMMAND_SUPPORT=y
CONFIG_AT_BT_COMMAND_SUPPORT=
CONFIG_AT_FS_COMMAND_SUPPORT=
CONFIG_AT_ETHERNET_SUPPORT=
CONFIG_AT_EAP_COMMAND_SUPPORT=
CONFIG_AT_OTA_SERVER_IP="iot.espressif.cn"
CONFIG_AT_OTA_SERVER_PORT=80
CONFIG_AT_OTA_TOKEN_KEY="dd93253c287f725de50d4071a05dd28b72056ca7"
CONFIG_AT_OTA_SSL_SUPPORT=y
CONFIG_AT_OTA_SSL_SERVER_IP="iot.espressif.cn"
CONFIG_AT_OTA_SSL_SERVER_PORT=443
CONFIG_AT_OTA_SSL_TOKEN_KEY="dd93253c287f725de50d4071a05dd28b72056ca7"
CONFIG_ESP_AT_FW_VERSION=""
 
```

I saw #101 and #51 and I'am sure that my terminal sends  0x0D 0x0A when I hit enter.

What do I do wrong?

**Comment by carlessole** [comment]
I have the same issue.
I see that the code that initialize the AT UART has been changed with this new release. Maybe now, UARTS can not be switched.
Any solution?
Thank you.

**Comment by xcguang** [comment]
You can refer to [https://github.com/espressif/esp32-at/blob/master/docs/ESP32_AT_Factory_Parameter_Bin.md](https://github.com/espressif/esp32-at/blob/master/docs/ESP32_AT_Factory_Parameter_Bin.md), we will remove configure part pin in menuconfig and set them in `components/customized_partitions/raw_data/factory_param/factory_param_data.csv`.
Currently, if you want to configure them by `make menuconfig`,  you need flash all binary files besides factory_param_xxx.bin

**Comment by xcguang** [comment]
[QCOM.zip](https://github.com/espressif/esp32-at/files/2729350/QCOM.zip)
@TomaszKosecki Please try it using this serial tool

**Comment by carlessole** [comment]
@xcguang thanks for the response.
So, to ensure that I understand you, are you suggesting that manually modify this file: `components/customized_partitions/raw_data/factory_param/factory_param_data.csv`?
And also that in future releases, it will be NOT possible configure AT uart through menuconfig right?

**Comment by xcguang** [comment]
@carlessole Yes, you are right. For the same chip, we want to make all module firmware the same, the differences are reflected in configuration files. Perhaps this is more convenient for user to maintain multiple modules, which will be mass produced.

**Comment by TomaszKosecki** [comment]
OK, now it's working.

It had nothing to do with serial tool I was using (Tera Term is perfectly ok).
I was not aware of the fact that "Factory param data" overwrites setting from "menuconfig" 
(maybe it whould be a good idea to mention about this in README.md?)

If the idea is to be able to change uart settings without "make menuconfig" and without recompilation
then what about "uart port number" in factory_param_data.csv ?

Now if I want to change uart from UART2 (which is default) to UART0 I need to change setings in both:
"make menuconfig" and factory_param_data.csv. Right? Or maybe I am missing something?

**Comment by xcguang** [comment]
@TomaszKosecki 
Yes, you are right. Maybe adding "uart port number" in factory_param_data.csv is a good idea. 
In fact, you also can think of AT UART port as fixed, and just change the PIN number in factory_param_data.csv.  For example, if you want to print log via uart with the same as AT port, you can change the log uart configuration in `UART for console output` as same as AT port.

**Comment by Funky185540** [comment]
Would you mind maybe adding this to the documentation i.e. in the "ESP32 AT instruction set and examples.pdf"? Because on page 2/97 there it clearly says: 

> Users can change to use another UART for AT communication. For example, if you want to use
UART0 for AT communication, you need to:
> - make menuconfig -> component config -> AT -> "AT UART settings" to set it to use UART 0


Took me 3 days to find this issue, that was really frustrating.

**Comment by Masi2017** [comment]
Hi @TomaszKosecki ,
I have the same problem but although I have changed the factory file still it's not working. Do I need to flash the factory_param_data individually? How can I do that?
Also I should mention that I have removed those parts that overwrites the uart setting fromfactory_param_data file but still it doesn't respond me . Also I have moved debug port to UART1 in menuconfig.
![image](https://user-images.githubusercontent.com/34062463/58348638-2fade900-7e2f-11e9-9552-d5c5252d7d9f.png)

**Comment by TomaszKosecki** [comment]
* Which UART do you want to use for AT commands? I think I read somewhere that UART1 pins are used internally on WROOM module for communication with FLASH.
* I didn't flash factory_param_data individually, just used "make -j8 flash"
* Settings which work for me for UART0:
   sdkconfig (edited with: make menuconfig):
     CONFIG_AT_UART_PORT=0
     CONFIG_AT_UART_PORT_RX_PIN=3
     CONFIG_AT_UART_PORT_TX_PIN=1
   components/customized_partitions/raw_data/factory_param/factory_param_data.csv
     WROOM-32,0xfcfc,1,1,1,1,13,CN,115200,1,3,22,19
* Settings which work for me for UART2:
   sdkconfig (edited with: make menuconfig):
     CONFIG_AT_UART_PORT=2
     CONFIG_AT_UART_PORT_RX_PIN=16
     CONFIG_AT_UART_PORT_TX_PIN=17
   components/customized_partitions/raw_data/factory_param/factory_param_data.csv
     WROOM-32,0xfcfc,1,1,1,1,13,CN,115200,17,16,15,14
* It looks like TX is working because you got "ready" in terminal. 
 
 
W dniu 2019-05-24 20:21:39 użytkownik Mm <notifications@github.com> napisał:
Hi @TomaszKosecki ,
I have the same problem but although I have changed the factory file still it's not working. Do I need to flash the factory_param_data individually? How can I do that?
Also I should mention that I have removed those parts that overwrites the uart setting fromfactory_param_data file but still it doesn't respond me . Also I have moved debug port to UART1 in menuconfig.

**Comment by Masi2017** [comment]
@TomaszKosecki Thanks for your response.
I want to use UART0, I did the same changes in sdkconfig and factory_param_data.csv but didn't work for me. Also yesterday I noticed that source code has been changed and uart hardware configuration for AT port has been removed from menuconfig! Now the only way the UART port can be changed is from source code ( still I can change RX and Tx pins from factory_param_data.csv ).
